src/sectionproperties/analysis/fea.py

```diff
@@ -845,23 +856,34 @@ def gauss_points(n: int) -> np.ndarray:
             ],
             dtype=float,
         )
-    else:
-        raise ValueError("n must be 1, 3 or 6.")
-
 
-def shape_function(
-    coords: np.ndarray,
-    gauss_point: tuple[float, float, float, float],
+    raise ValueError("n must be 1, 3, 4 or 6.")
+
+
+@lru_cache(maxsize=None)
+def __shape_function_cached(
+    coords: tuple[
```

Comment:
Sorry if I was unclear - because this is so long, I would prefer `coords: tuple[float, ...],` for readability :) The length is correctly specified in the docstring.